gtsam/geometry/Gal3.h

```diff
@@ -0,0 +1,249 @@
+
+  /// Named constructor from Pose3, velocity, and time with derivatives
+  static Gal3 FromPoseVelocityTime(const Pose3& pose, const Velocity3& v, double t,
+                                OptionalJacobian<10, 6> H1 = {},
+                                OptionalJacobian<10, 3> H2 = {},
+                                OptionalJacobian<10, 1> H3 = {});
+
+  /// @}
+  /// @name Component Access
+  /// @{
+
+  /// Access rotation component (Rot3)
+  const Rot3& rotation(OptionalJacobian<3, 10> H = {}) const;
+
+  /// Access translation component (Point3)
+  const Point3& translation(OptionalJacobian<3, 10> H = {}) const;
+
+  /// Access velocity component (Vector3)
+  const Velocity3& velocity(OptionalJacobian<3, 10> H = {}) const;
+
+  /// Access time component (double)
+  const double& time(OptionalJacobian<1, 10> H = {}) const;
+
+  // Convenience accessors matching NavState names
+  const Rot3& attitude(OptionalJacobian<3, 10> H = {}) const { return rotation(H); }
+  const Point3& position(OptionalJacobian<3, 10> H = {}) const { return translation(H); }
+
+  /// @}
+  /// @name Derived quantities
+  /// @{
+
+  /// Return rotation matrix (Matrix3)
+  Matrix3 R() const { return R_.matrix(); }
+
+  /// Return position as Vector3
+  Vector3 r() const { return Vector3(r_); } // Conversion from Point3
+
+  /// Return velocity as Vector3
+  const Vector3& v() const { return v_; }
+
+  /// Return time scalar
+  const double& t() const { return t_; }
+
+  /// Return 5x5 homogeneous matrix representation
+  Matrix5 matrix() const;
+
+  /// @}
+  /// @name Testable
+  /// @{
+
+  /// Output stream operator
+  GTSAM_EXPORT
+  friend std::ostream &operator<<(std::ostream &os, const Gal3& state);
+
+  /// Print with optional string prefix
+  void print(const std::string& s = "") const;
+
+  /// Check equality within tolerance
+  bool equals(const Gal3& other, double tol = 1e-9) const;
+
+  /// @}
+  /// @name Group
+  /// @{
+
+  /// Return the identity element
+  static Gal3 Identity() { return Gal3(); }
+
+  /// Return the inverse of this element
+  Gal3 inverse() const;
+
+  // Bring LieGroup::inverse() into scope (version with derivative)
+  using LieGroup<Gal3, 10>::inverse;
+
+  /// Group composition operator
+  Gal3 operator*(const Gal3& other) const;
+
+  /// @}
+  /// @name Group Action
+  /// @{
+
+  /**
+   * Apply Galilean transform to a spacetime Event
+   * @param e Input event (location, time)
+   * @param Hself Optional Jacobian wrt this Gal3 element's tangent space
+   * @param He Optional Jacobian wrt the input event's tangent space
+   * @return Transformed event
+   */
+  Event act(const Event& e, OptionalJacobian<4, 10> Hself = {},
+            OptionalJacobian<4, 4> He = {}) const;
+
+  /// @}
+  /// @name Lie Group Static Functions
+  /// @{
+
+  /// The type of the Lie algebra (matrix representation)
+  using LieAlgebra = Matrix5;
+
+  // Helper functions for accessing tangent vector components
```

Comment:
I might put these in an anonymous namespace in the .cpp file…

Reply:
Unless we need them here, can we move them to anonymous namespace in the .cpp file?